### PR TITLE
Implement CLI styler options overrides

### LINKED ISSUE #244: XamlStyler.Console should be able to accept options via command line arguments
Hey there. Thanks for adding XamlStyler.Console to NuGet, I've integrated it with my project and now my XAML files are formatted on every build. I'm no longer bound to Visual Studio and can now use Rider, yay. Good job!

Anyway, it would be really nice if the console tool would also accept individual formatting settings as command line options. Currently it accepts a path to a configuration file (or infers it automatically) but it's not super convenient if you just want to set 1 option.

I like how [Prettier did it](https://prettier.io/docs/en/options.html), they have default settings, configuration file and command line arguments, and the priority goes like this (first to last):

Command line arguments -> Config file -> Default settings.

That means both config file and command line arguments can coexist and there's no need for some extra validation.

If this issue is approved and there's no one willing to take it, I can probably make a PR to implement this.

### PR DESCRIPTION
**Pull request by Tyrrrz**
### Description:

Implement CLI overrides for styler options.
Changes to documentation will be needed to make note of this.

Fixes #244

### Checklist:
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] My changes generate no new warnings
* [ ] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [ ] I have tested my changes by running the extension in VS2017
* [ ] I have tested my changes by running the extension in VS2019
* [x] If changes to the [documentation](https://github.com/Xavalon/XamlStyler/wiki) are needed, I have noted this in the description above

### CODE REVIEW

File: XamlStyler.Console/XamlStylerConsole.cs [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+                stylerOptions.KeepFirstAttributeOnSameLine = options.KeepFirstAttributeOnSameLine.Value;
+
+            if (options.MaxAttributeCharactersPerLine != null)
+                stylerOptions.MaxAttributeCharactersPerLine = options.MaxAttributeCharactersPerLine.Value;
+
+            if (options.MaxAttributesPerLine != null)
+                stylerOptions.MaxAttributesPerLine = options.MaxAttributesPerLine.Value;
+
+            if (options.NoNewLineElements != null)
+                stylerOptions.NoNewLineElements = options.NoNewLineElements;
```
Comment (grochocki) [inline XamlStyler.Console/XamlStylerConsole.cs]: For the non-nullable types, add brackets {}

File: XamlStyler.Console/XamlStylerConsole.cs [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+                stylerOptions.ThicknessAttributes = options.ThicknessAttributes;
+
+            if (options.CommentSpaces != null)
+                stylerOptions.CommentSpaces = options.CommentSpaces.Value;
+        }
+
+        public void Process(ProcessType processType)
+        {
+            int successCount = 0;
+
```
Comment (grochocki) [inline XamlStyler.Console/XamlStylerConsole.cs]: nit: Remove extra line here

File: XamlStyler.Console/Options.cs [hunk 3]
```diff
@@ -0,0 +1,117 @@
+﻿using System.Collections.Generic;
```
Comment (grochocki) [inline XamlStyler.Console/Options.cs]: Can you add our standard header to the top of all code files?

"// © Xavalon. All rights reserved."

File: XamlStyler.Console/XamlStylerConsole.cs [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+            }
+
+            ApplyOptionOverrides(options, stylerOptions);
+
+            this.stylerService = new StylerService(stylerOptions);
+        }
+
+        private void ApplyOptionOverrides(Options options, IStylerOptions stylerOptions)
+        {
+            if (options.IndentSize != null)
```
[the remark this answers is not in the document]
Comment (Tyrrrz) [reply XamlStyler.Console/XamlStylerConsole.cs]: You can't do `options.IndentSize?.Value`. Plus the settings shouldn't be set to a new value unless an override specified.

File: XamlStyler.Console/XamlStylerConsole.cs (hunk 1 above)
Comment (Tyrrrz) [reply XamlStyler.Console/XamlStylerConsole.cs]: Not sure what you mean, they're all nullable :)
Do you want brackets on the `if` body?

File: XamlStyler.Console/XamlStylerConsole.cs (hunk 2 above)
Comment (Tyrrrz) [reply XamlStyler.Console/XamlStylerConsole.cs]: Done

File: XamlStyler.Console/Options.cs (hunk 3 above)
Comment (Tyrrrz) [reply XamlStyler.Console/Options.cs]: Done

File: XamlStyler.Console/XamlStylerConsole.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (grochocki) [reply XamlStyler.Console/XamlStylerConsole.cs]: Doh 🤦‍♂️ Of course, thansk!

File: XamlStyler.Console/XamlStylerConsole.cs (hunk 1 above)
Comment (grochocki) [reply XamlStyler.Console/XamlStylerConsole.cs]: > Do you want brackets on the if body?

Yep

**Comment by Tyrrrz** [comment]
Thanks :)

**Pull request merged**